### PR TITLE
Join table

### PR DESCRIPTION
**Pull request by ahzamir**
### In this pull request I have used JOIN for joining tables with many to many relationship according to these requirements:

- I have created a table named vets with the following columns:
  - id: integer (set it as autoincremented PRIMARY KEY)
  - name: string
  - age: integer
  - date_of_graduation: date
- There is a many-to-many relationship between the tables species and vets: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. I created a "join table" called specializations to handle this relationship.
- There is a many-to-many relationship between the tables animals and vets: an animal can visit multiple vets and one vet can be visited by multiple animals. I created a "join table" called visits to handle this relationship, it also keeps track of the date of the visit.
- I inserted the following data for vets:
  - Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
  - Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
  - Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
  - Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
  - Insert the following data for specialties:
  - Vet William Tatcher is specialized in Pokemon.
  - Vet Stephanie Mendez is specialized in Digimon and Pokemon.
  - Vet Jack Harkness is specialized in Digimon.
- I inserted the following data for visits:
  - Agumon visited William Tatcher on May 24th, 2020.
  - Agumon visited Stephanie Mendez on Jul 22th, 2020.
  - Gabumon visited Jack Harkness on Feb 2nd, 2021.
  - Pikachu visited Maisy Smith on Jan 5th, 2020.
  - Pikachu visited Maisy Smith on Mar 8th, 2020.
  - Pikachu visited Maisy Smith on May 14th, 2020.
  - Devimon visited Stephanie Mendez on May 4th, 2021.
  - Charmander visited Jack Harkness on Feb 24th, 2021.
  - Plantmon visited Maisy Smith on Dec 21st, 2019.
  - Plantmon visited William Tatcher on Aug 10th, 2020.
  - Plantmon visited Maisy Smith on Apr 7th, 2021.
  - Squirtle visited Stephanie Mendez on Sep 29th, 2019.
  - Angemon visited Jack Harkness on Oct 3rd, 2020.
  - Angemon visited Jack Harkness on Nov 4th, 2020.
  - Boarmon visited Maisy Smith on Jan 24th, 2019.
  - Boarmon visited Maisy Smith on May 15th, 2019.
  - Boarmon visited Maisy Smith on Feb 27th, 2020.
  - Boarmon visited Maisy Smith on Aug 3rd, 2020.
  - Blossom visited Stephanie Mendez on May 24th, 2020.
  - Blossom visited William Tatcher on Jan 11th, 2021.
- I write queries to answer the following:
  - Who was the last animal seen by William Tatcher?
  - How many different animals did Stephanie Mendez see?
  - List all vets and their specialties, including vets with no specialties.
  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
  - What animal has the most visits to vets?
  - Who was Maisy Smith's first visit?
  - Details for most recent visit: animal information, vet information, and date of visit.
  - How many visits were with a vet that did not specialize in that animal's species?
  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.

- I provided some screenshots from action results of my queries:
pro 4 1
pro 4 2

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -23,6 +23,26 @@ CREATE TABLE species (
     name TEXT NOT NULL
 );
 
+CREATE TABLE vets (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    name TEXT NOT NULL,
```
Comment (ShahierNashaat) [inline schema.sql:28]: - [x] Kindly be noticed that it is better to make the name varchar(255) as it is much smaller than TEXT.

File: data.sql [hunk 2]
```diff
@@ -29,4 +29,38 @@ UPDATE animals SET owner_id = 1 WHERE name = 'Agumon';
...
 UPDATE animals SET owner_id = 3 WHERE name = 'Devimon' OR name = 'Plantmon';
 UPDATE animals SET owner_id = 4 WHERE name = 'Charmander' OR name = 'Squirtle' OR name = 'Blossom';
-UPDATE animals SET owner_id = 5 WHERE name = 'Angemon' OR name = 'Boarmon';
\ No newline at end of file
+UPDATE animals SET owner_id = 5 WHERE name = 'Angemon' OR name = 'Boarmon';
+
+INSERT INTO vets (name, age, date_of_graduation)
+VALUES ('William Tatcher', 45, 'Apr 23, 2000'),
+('Maisy Smith', 26, 'Jan 17, 2019'),
+('Stephanie Mendez', 64, 'May 04, 1981'),
```
Comment (ShahierNashaat) [inline data.sql:37]: - [x] Kindly be noticed that the date of graduation for Stephanie Mendez is `May 17, 1981`.

File: data.sql [hunk 3]
```diff
@@ -29,4 +29,38 @@ UPDATE animals SET owner_id = 1 WHERE name = 'Agumon';
...
+('Vet Jack Harkness', 38, 'Jun 08, 2008');
```
Comment (ShahierNashaat) [inline data.sql:38]: - [x] Kindly be noticed that the name is just `Jack Harknes`.

File: queries.sql [hunk 4]
```diff
@@ -21,4 +21,13 @@ SELECT animals.name, full_name FROM animals RIGHT JOIN owners ON owners.id = own
 SELECT COUNT(*), species.name FROM animals JOIN species ON species_id = species.id GROUP BY species.name;
 SELECT animals.name FROM animals JOIN species ON species_id = species.id JOIN owners ON owner_id = owners.id WHERE species_id = 2;
 SELECT animals.name FROM animals JOIN owners ON owner_id = owners.id WHERE owners.id = 5 AND escape_attempts = 0;
-SELECT COUNT(*) as total, owners.id, owners.full_name FROM animals RIGHT JOIN owners ON owners.id = owner_id GROUP BY owners.id ORDER BY total DESC LIMIT 1;
\ No newline at end of file
+SELECT COUNT(*) as total, owners.id, owners.full_name FROM animals RIGHT JOIN owners ON owners.id = owner_id GROUP BY owners.id ORDER BY total DESC LIMIT 1;
+SELECT animals.name FROM animals JOIN visits ON animals_id = animals.id WHERE vets_id = 1 ORDER BY visits.date LIMIT 1;
+SELECT COUNT(animals.name) FROM animals JOIN visits ON animals_id = animals.id WHERE vets_id = 3;
```
Comment (ShahierNashaat) [inline queries.sql:27]: - [ ] Kindly be noticed that you should use distinct in this query as it is asking for the different animals that Stephanie Mendez sees.

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @ahzamir, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShahierNashaat — APPROVED** [review]
Hi @ahzamir , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahzamir** [comment]
@ShahierNashaat  Thank you very much for your kind review and feedback.

**Pull request merged**